src/one/util/ideaplugin/screenshoter/SaveImageAction.java

```diff
@@ -0,0 +1,69 @@
+package one.util.ideaplugin.screenshoter;
+
+import com.intellij.notification.NotificationDisplayType;
+import com.intellij.notification.NotificationGroup;
+import com.intellij.openapi.actionSystem.*;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.MessageType;
+import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.util.SystemProperties;
+import org.jetbrains.annotations.NotNull;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+
+public class SaveImageAction extends AnAction {
+
+    private static final NotificationGroup NOTIFICATION_GROUP = new NotificationGroup(
+            "image.saved.id",
+            NotificationDisplayType.STICKY_BALLOON,
+            false,
+            null
+    );
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent) {
+        Editor editor = CopyImageAction.geEditor(anActionEvent);
+        if (editor == null) return;
+
+        BufferedImage image = new ImageBuilder(editor).createImage();
+        saveImage(editor, image);
+    }
+
+
+    private void saveImage(@NotNull Editor editor, @NotNull BufferedImage image) {
+        Project project = editor.getProject();
+        CopyImageOptionsProvider.State options = CopyImageOptionsProvider.getInstance(project).getState();
+        String toSave = options.myDirectoryToSave;
+        if (StringUtil.isEmpty(toSave)) {
+            toSave = SystemProperties.getUserHome();
+        }
+        try {
+            SimpleDateFormat dt = new SimpleDateFormat("yyyyymmdd_hhmmss");
+            String date = dt.format(new Date());
+
+            File outFile = new File(FileUtil.toSystemDependentName(toSave + "/" + "Shot_" + date + ".png"));
+            FileUtil.createParentDirs(outFile);
+
+            ImageIO.write(image, "png", outFile);
+
+            NOTIFICATION_GROUP.createNotification("Image \n" + outFile.getPath() + "\nwas saved",
+                    MessageType.INFO).notify(project);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
I think this will end up in "Report to JetBrains" dialog which is pretty bad if user just supplied an invalid directory path or readonly location or just run out of disk space. Can we display a user-friendly notification like `"Cannot save "+path+":\n"+e.getMessage()`?

Reply:
:+1: